[카프카] API 테스트/문서 자동화 미션 제출합니다.

include42 opened the pull request:
안녕하세요. 카프카입니다.

미션 구현하여 PR 보냅니다.

감사합니다.

hotheadfactory committed:
feat: 문서화 테스트, AuthAcceptanceTest 구현

include42 committed:
feat: AuthAcceptanceTest 내의 session을 통한 로그인 테스트 구현

include42 committed:
feat: AuthAcceptanceTest 내의 bearer를 통한 로그인 테스트 구현

hotheadfactory committed:
feat: Bearer를 제외한 나머지 Interceptor 삭제

hotheadfactory committed:
feat: 문서 자동화 구현

include42 committed:
feat: MemberControllerTest에 update에 대한 테스트 추가, 문서자동화 템플릿에 내용 추가

include42 committed:
feat: MemberControllerTest의 update 테스트에 Token 생성 로직 추가

include42 committed:
feat: MemberController에 대한 문서 자동화 구현

api-guide MemberController의 CRUD 기능에 대한 템플릿 구성
MemberController Generic 변경
MemberControllerTest 컨트롤러의 CRUD 기능에 대한 문서 자동화 코드 구현

hotheadfactory committed:
feat: 회원가입 기능 프론트엔드 연결

- API 작성
- 빈칸 및 형식 예외 처리
- 에러 메시지 작성

include42 committed:
feat: frontend login 구현

Join.js join.js에서 이름 변경
Login.js Local Storage를 이용하여 JWT 관리

hotheadfactory committed:
feat: 마이페이지 기능 프론트엔드 연결

- 토큰으로 회원 정보를 불러오는 API 작성

hotheadfactory committed:
fix: AuthAcceptanceTest 수정

- /me/bearer -> /myinfo, /oauth/token -> /login 경로 수정 반영

include42 committed:
feat: frontend 회원탈퇴 기능 구현

MypageEdit.js 회원정보 수정에 대한 JS 구현
MemberController delete 시 id를 파라미터로 보내지 않도록 변경
BearerAuthInterceptor WebMvcConfig에서 연결되는 내용을 분기를 만들어 필터링함. 추후 수정 예정

include42 committed:
feat: frontend 회원정보 수정 기능 구현

MypageEdit.js 회원정보 수정에 대한 JS 구현
MemberController update 시 id를 파라미터로 보내지 않도록 변경

hotheadfactory committed:
feat: ErrorController 구현 및 올바른 HTTP 코드(400) 반환, 프론트엔드에서 에러 표시 구현

include42 committed:
feat: 사이드 케이스에 대한 백엔드에서의 검증 구현

member.dto 프론트에서 검증한 예외에 대해 재검증
 - Login 시 빈 이메일, 빈 암호, 잘못된 형식의 이메일이 전송될 경우 예외 발생
 - 회원가입 시 빈 이메일, 빈 이름, 빈 암호, 잘못된 형식의 이메일이 전송될 경우 예외 발생
 - 회원정보 수정 시 빈 이름, 빈 암호가 전송될 경우 예외 발생
 MemberService 회원 정보에 대한 검증 중 오류 시 예외 발생
 ExceptionController @Valid를 통한 예외 발생 시 메세지를 ErrorResponse에 담아 프론트에 전달

hotheadfactory committed:
feat: 로그인과 내 정보 보기에 대한 문서 자동화와 단위 테스트 구현

include42 committed:
feat: 문서 자동화를 위한 설정값 조정

include42 committed:
refactor: front에서 에러 발생 시의 디폴트 메세지 로직 추가, 로그인 세션 관련 에러 시 리다이렉션 기능 추가

include42 committed:
docs: README 작성

hotheadfactory committed:
fix: 인증 토큰을 제공하여도 로그인 페이지로 리다이렉트되는 오류 수정

include42 committed:
feat: data.sql fixture 추가를 통해 DB 초기화 동작 수행

hotheadfactory committed:
feat: 메인 페이지에 즐겨찾기 목록 버튼 추가

hotheadfactory committed:
feat: Favorite 관련 인수 테스트, 클래스 준비와 DB 설계 추가

include42 committed:
feat: FavoriteController의 Mock 작동 설계 및 DTO 클래스 구현

hotheadfactory committed:
feat: Favorite 추가, 삭제 구현 완료

hotheadfactory committed:
feat: Favorite에 대한 서비스 로직 구현 완료, 테스트 통과

FavoriteAcceptanceTest Favorite에 대한 인수테스트 구현
FavoriteController Favorite에 대한 컨트롤러 기능 수행하도록 구현
FavoriteService Favorite에 대한 서비스 로직 구현
Favorites Favorite의 일급 컬렉션으로써 FavoriteResponse의 List를 반환하는 비즈니스 로직 구현

hotheadfactory committed:
feat: Favorite 삭제 기능 프론트엔드에 구현

hotheadfactory committed:
feat: Favorite 조회와 별 토글 기능 구현

- 겸색결과를 로딩할 시에도 현재 즐겨찾기 등록 상태를 식별할 수 있도록 함
- 별 아이콘 토글로 바로 즐겨찾기에 추가하거나 삭제할 수 있음

hotheadfactory committed:
feat: rest docs 문서 자동화 테스트 작성

hotheadfactory committed:
feat: 테스트 통과하지 못하는 문제 수정

- data.sql 테스트에서 import하지 않도록 수정

include42 committed:
feat: 문서 자동화 템플릿 작성 완료

include42 committed:
feat: Station 관련 예외 처리 추가 및 README 갱신

hotheadfactory committed:
feat: 누락된 Member 테스트 구현

- 즐겨찾기의 추가/삭제

hotheadfactory committed:
docs: README.md 진행 상황 반영

include42 committed:
Merge pull request #1 from hotheadfactory/step2-favorite

Step2 favorite version PR

include42 committed:
refactor: FavoriteService가 StationService를 주입받도록 변경

include42 committed:
feat: FavoriteService에 대한 테스트 코드 작성

include42 committed:
docs: 진행사항 README에 반영

kang-hyungu commented on src/main/java/wooteco/subway/domain/member/Favorite.java:
```diff
@@ -0,0 +1,38 @@
+    private Long sourceId;
+    private Long destinationId;
+
+    public Favorite() {
```
```suggestion
    private Favorite() {
```
기본생성자를 사용하지 않으면 private으로 막아주세요
다른 도메인 클래스도 마찬가지로 적용해주세요

kang-hyungu commented on src/main/java/wooteco/subway/domain/member/Favorites.java:
```diff
@@ -0,0 +1,62 @@
+                .collect(Collectors.toList());
+    }
+
+    private FavoriteResponse toFavoriteResponse(List<Station> stations, Favorite favorite) {
```
List<Station>를 일급 컬렉션인 Sations로 대체하면 코드 복잡도가 줄어들겠네요

kang-hyungu commented on src/main/java/wooteco/subway/domain/member/Favorites.java:
```diff
@@ -0,0 +1,62 @@
+
+    private FavoriteResponse toFavoriteResponse(List<Station> stations, Favorite favorite) {
+        String sourceName = stations.stream()
+                .filter(station -> station.getId().equals(favorite.getSourceId()))
```
Station 객체에서 getter로 값을 꺼내서 비교하지 말고 메서드를 추가하는게 좀더 객체지향스럽겠네요

kang-hyungu commented on src/main/java/wooteco/subway/domain/member/Favorites.java:
```diff
@@ -0,0 +1,62 @@
+        favorites.remove(favorite);
+    }
+
+    public List<FavoriteResponse> toFavoriteResponses(List<Station> stations) {
```
도메인 클래스에 dto를 추가하지 않는게 좋습니다.
다양한 형태의 dto가 추가될 때 마다 도메인 클래스를 변경해야 되고 도메인이 다른 레이어의 객체에게 의존하는 문제도 생깁니다

include42 replied:
이 부분에 대해서는 생각하지 못했네요!! DTO 개념에 대해서도 다시 배우는 계기가 된 것 같습니다.

include42 replied:
그렇다면, 반대로 DTO 내에서 Favorites를 받아 List<FavoriteResponse>를 반환하는 static method가 있는 건 괜찮을까요?
DTO 클래스에 getter/setter 외의 로직이 어느정도 있어도 괜찮은지...에 대해 아직 확신이 없어서 1급컬렉션에 로직을 넣는 것으로 구현했었던지라... 고민이 됩니다

kang-hyungu replied:
지금 단계에서는 문제 없다고 생각합니다
나중에 좀더 복잡해지면 mapper를 활용하는 방법도 있습니다.

kang-hyungu commented on src/main/java/wooteco/subway/domain/member/Favorites.java:
```diff
@@ -0,0 +1,62 @@
+        return new FavoriteResponse(sourceName, destinationName);
+    }
+
+    public Optional<Favorite> findById(Long sourceId, Long destinationId) {
```
```suggestion
    public boolean existsFavorite(Long sourceId, Long destinationId) {
```
optional로 반환할 필요는 없겠네요

kang-hyungu commented on src/main/java/wooteco/subway/domain/member/Favorites.java:
```diff
@@ -0,0 +1,62 @@
+    public Optional<Favorite> findById(Long sourceId, Long destinationId) {
+        return favorites.stream()
+                .filter(favorite -> favorite.getSourceId().equals(sourceId) && favorite.getDestinationId().equals(destinationId))
+                .findFirst();
```
anymatch를 활용해보아요

kang-hyungu commented on src/main/java/wooteco/subway/service/member/FavoriteService.java:
```diff
@@ -0,0 +1,60 @@
+
+@Service
+public class FavoriteService {
+    private MemberRepository memberRepository;
```
멤버 변수에 final을 붙여서 이 서비스에 꼭 필요한 객체라는 표시를 해주세요

kang-hyungu commented on src/main/java/wooteco/subway/service/member/FavoriteService.java:
```diff
@@ -0,0 +1,60 @@
+        this.stationService = stationService;
+    }
+
+    public void createFavorite(FavoriteRequest favoriteRequest, Member member) {
```
cud 메서드에 트랜잭션 필수 입니다
이 메서드를 포함해 다른 메서드에 트랜잭션 적용해보아요

kang-hyungu replied:
다른 서비스 클래스에도 트랜잭션을 적용해보아요

kang-hyungu commented on src/main/java/wooteco/subway/service/member/FavoriteService.java:
```diff
@@ -0,0 +1,60 @@
+        validate(favoriteRequest);
+        Long sourceId = stationService.findStationByName(favoriteRequest.getSourceName()).getId();
+        Long destinationId = stationService.findStationByName(favoriteRequest.getDestinationName()).getId();
+        member.removeFavoriteById(sourceId, destinationId);
```
```suggestion
        member.removeFavorite(new Favorite(sourceId, destinationId));
```
member.addFavorite 메서드처럼 Favorite 객체로 메시지 주고 받는게 어떨까요?

kang-hyungu commented on src/main/java/wooteco/subway/service/member/FavoriteService.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    public List<FavoriteResponse> showAllFavorites(Member member) {
+        Favorites favorites = new Favorites(member.getFavorites());
```
member에서 getter로 값을 꺼내서 Favorites를 만들지 말고 member가 바로 Favorites를 반환해도 되지 않을까요?

kang-hyungu commented on src/main/java/wooteco/subway/service/member/FavoriteService.java:
```diff
@@ -0,0 +1,60 @@
+        return favorites.toFavoriteResponses(stationService.findStations());
+    }
+
+    public boolean ifFavoriteExist(FavoriteRequest favoriteRequest, Member member) {
```
```suggestion
    public boolean hasFavorite(FavoriteRequest favoriteRequest, Member member) {
```
boolean으로 반환하는 메서드는 is나 has를 써주면 좀더 의미가 잘 전달됩니다

include42 replied:
복잡하게 생각해서 오히려 전달이 잘 안됐네요!! 감사합니다 😂

kang-hyungu commented on src/main/java/wooteco/subway/service/member/FavoriteService.java:
```diff
@@ -0,0 +1,60 @@
+        return favorites.toFavoriteResponses(stationService.findStations());
+    }
+
+    public boolean ifFavoriteExist(FavoriteRequest favoriteRequest, Member member) {
```
```suggestion
    public boolean hasFavorite(FavoriteRequest favoriteRequest, Member member) {
        validate(favoriteRequest);
        Long sourceId = stationService.findStationByName(favoriteRequest.getSourceName()).getId();
        Long destinationId = stationService.findStationByName(favoriteRequest.getDestinationName()).getId();
        return member.hasFavorite(sourceId, destinationId);
    }
```
member 객체가 favorites 객체를 가지고 있으니 member 통해서 favorite이 있는지 알수 있겠네요
member 객체는 favorites의 메시지 결과를 단순 전달하는 역할만 하면 될거에요

kang-hyungu commented on src/main/java/wooteco/subway/service/member/MemberService.java:
```diff
@@ -22,7 +23,7 @@ public Member createMember(Member member) {
 
     public void updateMember(Long id, UpdateMemberRequest param) {
-        Member member = memberRepository.findById(id).orElseThrow(RuntimeException::new);
+        Member member = memberRepository.findById(id).orElseThrow(() -> new InvalidAuthenticationException("ILLEGAL_ACCESS"));
```
memberRepository.findById(id).orElseThrow(() -> new InvalidAuthenticationException("ILLEGAL_ACCESS"))
1. 이 부분을 메서드로 추출하면 좋겠네요
2. InvalidAuthenticationException가 적절해보이진 않네요. 컨트롤러에서 이미 인증이 끝났고 서비스 레이어에서는 멤버 번호로 디비 조회를 했는데 데이터가 없는 것이니 not found 같은 이름이 적절하지 않을까요?

kang-hyungu commented on src/main/java/wooteco/subway/service/member/MemberService.java:
```diff
@@ -32,20 +33,15 @@ public void deleteMember(Long id) {
 
     public String createToken(LoginRequest param) {
-        Member member = memberRepository.findByEmail(param.getEmail()).orElseThrow(RuntimeException::new);
+        Member member = memberRepository.findByEmail(param.getEmail()).orElseThrow(() -> new InvalidAuthenticationException("WRONG_EMAIL"));
```
memberRepository.findByEmail(param.getEmail()).orElseThrow(() -> new InvalidAuthenticationException("WRONG_EMAIL"));

1. 메서드 추출 가능해보이네요
2. 이 메서드의 예외 처리도 not found가 적절해보이네요

kang-hyungu commented on src/main/java/wooteco/subway/service/member/dto/FavoriteRequest.java:
```diff
@@ -0,0 +1,26 @@
+    public FavoriteRequest() {
+    }
+
+    public FavoriteRequest(@NotBlank(message = "EMPTY_SOURCE") String sourceName, @NotBlank(message = "EMPTY_DESTINATION") String destinationName) {
```
이미 멤버변수에 not blank 어노테이션 표시해줬으니 파라미터에는 안해도 되지 않나요?

kang-hyungu commented on src/main/java/wooteco/subway/service/path/PathService.java:
```diff
@@ -29,12 +30,12 @@ public PathService(StationRepository stationRepository, LineRepository lineRepos
         List<Line> lines = lineRepository.findAll();
-        Station sourceStation = stationRepository.findByName(source).orElseThrow(RuntimeException::new);
-        Station targetStation = stationRepository.findByName(target).orElseThrow(RuntimeException::new);
+        Station sourceStation = stationRepository.findByName(source).orElseThrow(WrongStationException::new);
```
stationRepository.findByName(source).orElseThrow(WrongStationException::new);
이 부분을 StationService에 메서드 추가해서 중복 제거하는게 좋겠네요

kang-hyungu commented on src/main/java/wooteco/subway/service/path/PathService.java:
```diff
@@ -29,12 +30,12 @@ public PathService(StationRepository stationRepository, LineRepository lineRepos
+        Station targetStation = stationRepository.findByName(target).orElseThrow(WrongStationException::new);
 
         List<Long> path = graphService.findPath(lines, sourceStation.getId(), targetStation.getId(), type);
         List<Station> stations = stationRepository.findAllById(path);
```
일급 컬렉션 Staions를 활용하면 코드 복잡도를 줄일 수 있겠네요

kang-hyungu commented on src/main/java/wooteco/subway/service/path/PathService.java:
```diff
@@ -59,7 +60,7 @@ private Station extractStation(Long stationId, List<Station> stations) {
+                .orElseThrow(WrongStationException::new);
     }
 
     private List<LineStation> extractPathLineStation(List<Long> path, List<LineStation> lineStations) {
```
이 메서드도 일급 컬렉션인 LineStations에서 처리해도 되지 않을까요?
서비스에 도메인 로직이 있으면 중복 코드 발생하기 쉬우니 도메인 클래스로 로직을 옮기는게 좋겠네요

kang-hyungu replied:
extractStation 메서드도 Staions 로 로직을 옮겨보아요

kang-hyungu commented on src/main/java/wooteco/subway/web/member/FavoriteController.java:
```diff
@@ -0,0 +1,49 @@
+    @GetMapping("/favorites/exists")
+    public ResponseEntity<Map<String, Boolean>> ifFavoriteExists(@LoginMember Member member, @RequestParam String source, @RequestParam String destination) {
+        Boolean ifExists = favoriteService.ifFavoriteExist(new FavoriteRequest(source, destination), member);
+        Map<String, Boolean> result = new HashMap<>();
```
dto로 만드는게 어떨까요

kang-hyungu reviewed (changes requested):
카프카 다시 만나서 반가워요 👋 
먼저 수정해보면 좋을 부분에 피드백 남겼어요
확인해보시고 궁금한 점 생기면 코멘트 남기거나 dm 주세요

include42 committed:
refactor: 기본 생성자의 접근 제어자 private로 수정

include42 committed:
refactor: Favorites 로직 수정 및 FavoriteResponse에 로직 추가

Favorites 코드 리팩토링 및 DTO 반환 로직 삭제
FavoriteResponse static 메서드를 통해 Stations와 Favorites를 받아 List<FavoriteResponse>를 반환하는 로직 추가

include42 committed:
refactor: Service 로직에 Transactional 어노테이션 추가

include42 committed:
refactor: 코드 가독성 증가를 위한 리팩토링, 예외 클래스 추가

MemberNotFoundException 인증이 된 Member를 DB에서 찾을 수 없을 때 발생하는 예외 클래스

include42 committed:
refactor: 요구사항 적용에 따른 리팩토링

PathService StationRepository 대신 StationService와 의존 관계 가지도록 주입
StationService findAllById 수행하는 메서드 구현
MemberService 중복 로직 제거
FavoriteRequest 어노테이션 실수 수정

include42 committed:
refactor: PathService의 Station, LineStation 에 대한 1급컬렉션 구현

include42 committed:
refactor: Line/LineStation Service 로직에 Transactional 추가

include42 committed:
feat: Favorite의 존재 여부를 전달하는 DTO 클래스 구현

ExistsResponse DTO 클래스로 내부에 boolean 값을 가지고 있음.
FavoriteController Favorite가 존재하는지 검사하는 로직 수행 시, 그 결과값을 ExistsResponse로 전달

include42 commented:
안녕하세요 구구!! 카프카입니다.
리뷰 반영이 많이 늦어졌네요 😢 
그래도 지적해주신 내용이 너무 좋아서 많이 공부할 수 있었습니다.

아직 부족한 부분이 많은 코드입니다.
혹시 잘못된 부분이 있다면 지적해주신다면 확인하겠습니다.
감사합니다!!

kang-hyungu commented on src/main/java/wooteco/subway/domain/line/LineStations.java:
```diff
@@ -67,10 +69,33 @@ private void updatePreStationOfNextLineStation(Long targetStationId, Long newPre
+        Set<LineStation> paths = new LinkedHashSet<>();
+
+        for (Long stationId : path) {
+            if (preStationId == null) {
```
2뎁스를 제거해보아요

kang-hyungu commented on src/main/java/wooteco/subway/service/member/dto/FavoriteResponse.java:
```diff
@@ -0,0 +1,41 @@
+                .collect(Collectors.toList());
+    }
+
+    private static FavoriteResponse toFavoriteResponse(Stations stations, Favorite favorite) {
```
이 메서드에 비즈니스 로직이 들어갔네요
도메인 클래스가 제공하는 기능을 dto에서 사용하면 안됩니다
서비스 클래스에서 domain -> dto로 변환하도록 바꾸면 어떨까요?

kang-hyungu commented on src/main/java/wooteco/subway/service/path/PathService.java:
```diff
@@ -1,87 +1,58 @@
         this.graphService = graphService;
     }
 
     public PathResponse findPath(String source, String target, PathType type) {
```
이 메서드가 복잡해 보이네요. 복잡한 로직을 리팩토링 해보아요
`List<Line> lines`를 일급 컬렉션으로 바꿔보는 것부터 시작해보면 좋겠네요

kang-hyungu commented on src/test/java/wooteco/subway/AcceptanceTest.java:
```diff
@@ -21,6 +25,7 @@
+@ActiveProfiles("test")
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @Sql("/truncate.sql")
 public class AcceptanceTest {
```
api가 늘어나면 AcceptanceTest 클래스에 계속 메서드가 추가 되겠네요
아래 샘플 코드처럼 get, post, put, delete 호출에 대한 중복 코드를 제거하면 AcceptanceTest가 변경될 일은 없지 않을까요?

```java
    <T> T post(String path, Map<String, String> params, Class<T> responseType) {
        return
                given().
                        body(params).
                        contentType(MediaType.APPLICATION_JSON_VALUE).
                        accept(MediaType.APPLICATION_JSON_VALUE).
                when().
                        post(path).
                then().
                        log().all().
                        statusCode(HttpStatus.CREATED.value()).
                        extract().as(responseType);
    }
```

kang-hyungu replied:
인수 테스트는 충분히 테스트 되고 있는데 도메인 클래스에 대한 테스트가 충분해 보이지 않네요
도메인 클래스에 대한 유닛 테스트도 추가해보면 좋겠어요

kang-hyungu reviewed (changes requested):
카프카, 피드백 반영 하신 내용 확인했어요
몇 가지 추가로 더 해보면 좋을 부분에 피드백 남겼어요
확인해보시고 궁금한 점 생기면 코멘트 남기거나 dm 주세요

include42 committed:
refactor: LineStations의 로직 간소화

include42 committed:
refactor: 요구사항에 따른 코드 리팩토링

GraphService 일급 컬렉션을 활용하도록 코드 리팩토링
PathService 일급 컬렉션을 활용하도록 코드 리팩토링, 메서드 분리 수행
FavoriteResponse 도메인 로직 제거
FavoriteService domain -> dto 변환 로직을 FavoriteResponse에서 옮겨옴

include42 committed:
debug: LineStations 내의 버그 수정

include42 committed:
refactor: Test 코드에 템플릿 적용, Create 수정

All create(post) 관련 로직 템플릿 사용한 범용 코드로 수정. 다만 반환값 차이, 토큰 차이 등으로 인한 미적용 코드가 남아있음. 추후 수정예정.

include42 committed:
refactor: Test 코드에 템플릿 적용, 중복 코드 제거

include42 commented:
안녕하세요. 카프카입니다.

프로젝트 준비가 갑자기 생겨서...경황이 없어 리뷰를 너무 늦게 보내드리네요 😢 
말씀해주신 내용 덕에 많이 학습할 수 있었습니다.
DTO에 대한 부분도 생각조차 못했고, 테스트 코드의 템플릿 사용 부분은.. 👍 👍 
아직 부족하지만 더 학습해보려고 합니다.

아무쪼록 늦은 리뷰요청 죄송합니다.
혹시 빠진 부분이 있다면 지적 부탁드립니다!!
감사합니다.

kang-hyungu commented on src/main/java/wooteco/subway/domain/line/LineStations.java:
```diff
@@ -77,25 +77,22 @@ public int getTotalDuration() {
 
     public LineStations extractPathLineStation(List<Long> path) {
-        Long preStationId = null;
+        Long preStationId = path.get(0);
```
path가 null이거나 empty면 NPE나 index out of range가 발생하겠네요
방어 코드를 추가해두면 좋을 것 같아요

kang-hyungu commented on src/main/java/wooteco/subway/service/path/PathService.java:
```diff
@@ -1,87 +1,58 @@
-                    .filter(it -> it.isLineStationOf(finalPreStationId, stationId))
-                    .findFirst()
-                    .orElseThrow(RuntimeException::new);
+        Stations pathStation = path.stream()
```
이 코드도 도메인 클래스로 옮길 수 있겠네요
Stations 클래스에 path를 인자로 받는 메서드를 추가하고 의미 있는 메서드명을 지어주면 읽기 좋은 코드가 될거에요

kang-hyungu commented on src/test/java/wooteco/subway/AcceptanceTest.java:
```diff
@@ -64,67 +64,97 @@ public static RequestSpecification given() {
                 log().all();
     }
 
+    public <T> T createByName(String path, String name, Class<T> responseType) {
```
AcceptanceTest는 get, post, put, delete에 해당되고 전체 인수 테스트에서 공통으로 사용할 메서드만 남기는게 좋겠네요
그 외 특정 인수테스트에서만 사용하는 메서드들은 해당 인수테스트로 옮겨주면 AcceptanceTest 클래스 복잡도가 줄어들거에요

kang-hyungu reviewed (approved):
카프카, 피드백 반영 확인했어요
피드백 구현하느라 수고하셨어요 👏 
추가로 남긴 코멘트는 다음 미션 때 참고하세요
pr은 머지 처리할게요!
